tsdb/engine/tsm1/file_store.go

```diff
@@ -595,9 +596,18 @@ func (f *FileStore) Open() error {
 		if res.r.LastModified() > lm {
 			lm = res.r.LastModified()
 		}
-
+		isEmpty = false
+	}
+	if isEmpty {
+		if fi, err := os.Stat(f.dir); err == nil {
+			f.lastModified = fi.ModTime().UTC()
```

Comment:
Does it need to account for tombstones? I'm not sure. TSMReader appears to: https://github.com/influxdata/influxdb/blob/master-1.x/tsdb/engine/tsm1/reader.go#L527

Reply:
I think it does account for tombstones.  In the loop [here](https://github.com/influxdata/influxdb/blob/bda3d2898219e1cdec1398b238b8bfcb5014aab8/tsdb/engine/tsm1/file_store.go#L596), `res.r.LastModfied()` checks the tombstones, so the directory modification time is only used if the `readerC` channel is completely empty.